src/main/java/com/twcable/grabbit/tools/monitor/JobStatus.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2014-2016 Time Warner Cable, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.twcable.grabbit.tools.monitor;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.yaml.snakeyaml.Yaml;
+
+import java.net.URI;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Map;
+import java.util.Objects;
+
+import static com.twcable.grabbit.tools.monitor.JobState.COMPLETED;
+import static com.twcable.grabbit.tools.monitor.JobState.FAILED;
+import static com.twcable.grabbit.tools.monitor.JobState.RUNNING;
+import static com.twcable.grabbit.tools.monitor.JobState.UNKNOWN;
+import static java.time.ZonedDateTime.now;
+import static java.util.Collections.emptyMap;
+
+/**
+ * An immutable value representing the status of a job, as returned by Grabbit's REST API.
+ */
+@SuppressWarnings({"unused", "WeakerAccess", "PMD.CyclomaticComplexity", "PMD.ModifiedCyclomaticComplexity",
+    "PMD.StdCyclomaticComplexity", "PMD.NPathComplexity", "checkstyle:VisibilityModifier"})
+public class JobStatus {
```

Comment:
Not using "autoconverters" like GSON because want the ability to have strong typing (instead of passing around `Map`s), a flattened datastructure (i.e., the "exitStatus" substructure), and use Java 8's really nice DateTime classes. By the time I'd set up the factories/transformers/etc to do that, it would be more code and more complex than doing it manually.